File: solutions/alien-language/dart/approach6.dart
/* Simple Hello, World! program */
void main(){
   var expresion  = "(zyx)bc";
	var counter  = 0;   
	expresion = expresion.replaceAll( '(', '[');
	expresion = expresion.replaceAll(')', ']');
	RegExp exp = new RegExp(expresion);
	var t = ["abc", "bca", "dac", "dbc", "cba"];
	Iterable<Match> matches ;
	for (var i = 0; i < t.length; i++) {
         matches=exp.allMatches(t[i]);
        counter+=matches.length;     
    }
    print(counter);


    
}

File: solutions/alien-language/scala/approach6.scala
object HelloWorld {
    import scala.util.matching.Regex
   def main(args: Array[String]) {
      
    var diccionario= Array("abc","bca","dac","dbc","cba")
    var expresion= "(zyx)bc"
    var counter =0
    expresion=expresion.replace('(','[')
    expresion=expresion.replace(')',']')
    val pattern = new Regex(expresion)
    for ( i<- 0 to diccionario.length-1) {
        
        if((pattern findAllIn diccionario(i)).exists(_.trim.nonEmpty)){
            counter=counter+1
        }
    }
    println(counter)
    
    
}
}


